pkg/network/handlers/work_package_submission.go

```
@@ -0,0 +1,133 @@
+}
+
+// NewWorkPackageSubmissionHandler creates a new handler instance with the given fetcher.
+func NewWorkPackageSubmissionHandler(fetcher ImportedSegmentsFetcher) *WorkPackageSubmissionHandler {
+	return &WorkPackageSubmissionHandler{
+		Fetcher: fetcher,
+	}
+}
+
+// HandleStream implements the guarantor side of the CE-133 protocol.
+// It expects two messages:
+//
+//	Message 1: [Core Index (u16) ++ work.Package]
+//	Message 2: [Extrinsic data (raw bytes)]
+//
+// After reading these it should fetch imported segments from the availability system
+func (h *WorkPackageSubmissionHandler) HandleStream(ctx context.Context, stream quic.Stream) error {
+	msg1, err := ReadMessageWithContext(ctx, stream)
+	if err != nil {
+		return fmt.Errorf("failed to read message 1: %w", err)
+	}
+
+	if len(msg1.Content) < 2 {
+		return fmt.Errorf("message is too short")
+	}
+
+	var coreIndex uint16
+	if err = jam.Unmarshal(msg1.Content[:2], &coreIndex); err != nil {
+		return fmt.Errorf("failed to unmarshal core index: %w", err)
+	}
+
+	var pkg work.Package
+	if err = jam.Unmarshal(msg1.Content[2:], &pkg); err != nil {
+		return fmt.Errorf("failed to unmarshal work package: %w", err)
+	}
+
+	if err = pkg.ValidateSize(); err != nil {
+		return fmt.Errorf("failed to validate work package: %w", err)
+	}
+
+	msg2, err := ReadMessageWithContext(ctx, stream)
+	if err != nil {
+		return fmt.Errorf("failed to read extrinsics message: %w", err)
+	}
+	extrinsics := msg2.Content
+
+	fmt.Printf("received submission with coreIndex=%d, work package with %d work items, extrinsics=%d bytes\n",
+		coreIndex, len(pkg.WorkItems), len(extrinsics))
+
+	for _, item := range pkg.WorkItems {
+		for _, imp := range item.ImportedSegments {
+			_, err = h.Fetcher.FetchImportedSegment(imp.Hash)
+			if err != nil {
+				// retry or reject package
+				continue
+			}
+			// Process or store segment data
+		}
+	}
+
+	if err = stream.Close(); err != nil {
+		fmt.Printf("failed to close stream: %d", err)
```

Should we return the error?

---

Also why `%d` ?

---

fixed